src/FeedForwardPid.cpp

```diff
@@ -6,27 +6,45 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+#include <limits>
+
 #include "iarc7_motion/FeedForwardPid.hpp"
 
 using namespace Iarc7Motion;
 
-FeedForwardPid::FeedForwardPid(double p_gain, double i_gain, double d_gain) : p_gain_(p_gain), i_gain_(i_gain), d_gain_(d_gain), setpoint_(0.0)
+FeedForwardPid::FeedForwardPid(double p_gain, double i_gain, double d_gain,
+                               double i_accumulator_max, double i_accumulator_min)
+    : p_gain_(p_gain),
+      i_gain_(i_gain),
+      d_gain_(d_gain),
+      setpoint_(0.0),
+      i_accumulator_(0.0),
+      last_current_value_(0.0),
+      i_accumulator_max_(i_accumulator_max),
+      i_accumulator_min_(i_accumulator_min)
 {
     // Nothing to do
 }
 
 double FeedForwardPid::update(double current_value, double time_delta)
 {
-
-    static double i_accumulator{0.0};
-    static double last_current_value{0.0};
-
+    if (time_delta == 0) {
+        time_delta = std::numeric_limits<double>::min();
```

Comment:
Hmmm. The real problem is that a if the input value changes and the time delta is zero than we have a step function on our input value which is impossible physically.

I think we should be erroring at this point and the caller should handle the error.

Otherwise the value we put out should be the result of discarding the results of the previous addition to the I_accumulator recalculating p. And recalculating d using the last last setpoint.